### PR TITLE
Update CODEOWNERS with more specific code owners

### LINKED ISSUE #1252: [FEATURE] Split `CODEOWNERS` to better identify actual code owners
**Is your feature request related to a problem?**
Our [current `CODEOWNERS` file](https://github.com/opensearch-project/dashboards-observability/blob/43a044fa5075eedf0ab4c232e39bd5b149c63751/.github/CODEOWNERS) currently marks every maintainer as an owner of the entire codebase. The issue is that code owners are automatically added as reviewers to PRs that touch owned code, which in practice means that every maintainer is assigned as a reviewer on every PR. This makes the review notifications very noisy and makes it harder to identify PRs where a maintainer is actually needed.

**What solution would you like?**
`CODEOWNERS` turns out to be capable of much more than a global mask, though. It supports [gitignore-like syntax](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax) to assign owners to individual files. By making better use of this file, we can get PRs to the people that need to see them, faster. For example, for adding the integrations project's main developers, we could consider adding the line:

```
**/integrations   @Swiddis @YANG-DB 
```

Which would add only the two mentioned devs as a reviewer, since `CODEOWNERS` only applies the last matching line in a file.

It also can just reference the [opensearch-project/observability team](https://github.com/orgs/opensearch-project/teams/observability) directly instead of listing every dev, which has the benefit of better dynamic management and also would help make the review process less noisy.

```
*   @opensearch-project/observability 
```

**What alternatives have you considered?**
There are a few other approaches to the issue.
- We could just disable automatically adding code owners in the repository review settings, making the process manual. I think this would still be better than adding everyone.
- We could use a more elaborate system like [assign-reviewer-by-label](https://github.com/buildsville/assign-reviewer-by-label) to implement this functionality through PR labels instead of CODEOWNERS.

**Do you have any additional context?**
This type of strategy is put to good use in some large projects like [CPython](https://github.com/python/cpython/blob/main/.github/CODEOWNERS) and [Node.js](https://github.com/v8/node/blob/main/.github/CODEOWNERS)

### PR DESCRIPTION
**Pull request by Swiddis**
### Description
Coming from #1252, this PR updates codeowners to:
- Include less owners overall, to get more focused reviews.
- Adds dedicated owners for the major subcomponents.

### Issues Resolved
Closes #1252.

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by YANG-DB — APPROVED** [review]

File: .github/CODEOWNERS
Comment (Swiddis) [inline .github/CODEOWNERS:2]: Looks like the team doesn't have write access, any workaround better than reverting?

**Comment by ps48** [comment]
I propose adding the code-owners as below:
Logs explorer: @mengweieric 
panels: @pjfitzgibbons 
app analytics: @paulstn 
notebooks: @ps48 
metrics: @kavithacm 
trace analytics: @RyanL1997 
Integrations: @Swiddis 
Datasources: @RyanL1997

**Review by kavithacm — APPROVED** [review]
Can you add Ryan as a maintainer before merging this PR?

**Review by derek-ho — CHANGES_REQUESTED** [review]
@Swiddis can we reach out to @pjfitzgibbons to see if he wants to be moved to emeritus prior to merging?

**Pull request closed without merging**
